### PR TITLE
Improved SRA tests, corrected error message

### PR DESCRIPTION
**Pull request by a-nikitiuk**
### Description

These changes are intended to resolve most of previously reported SRA-related issues:

- #442 - updated tests to use much smaller files, so that footprint by SRA cache on Mac OS X is about 30 Mb (on other OS issue does not exist, since we use sparse files there and footprint is even smaller)
- #459, #461 - now each network-related test checks if it can resolve an accession, and skips the test in case of failure
- #460, #545 - updated error message in SRAFileReader, so that it tells about a possibility of a connection problem

### Checklist

- [x] Code compiles correctly
- [x] New tests covering changes and new functionality
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary
- [x] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

File: src/main/java/htsjdk/samtools/SRAFileReader.java [hunk 1]
```diff
@@ -61,7 +61,8 @@ public SRAFileReader(final SRAAccession acc) {
         this.acc = acc;
 
         if (!acc.isValid()) {
-            throw new IllegalArgumentException("Invalid SRA accession was passed to SRA reader: " + acc);
+            throw new IllegalArgumentException("SRAFileReader: cannot resolve SRA accession '" + acc + "'\n" +
```
Comment (yfarjoun) [inline src/main/java/htsjdk/samtools/SRAFileReader.java]: isn't it clear at some point in the code that a connection has been established but that the accession is bad? why not inform the user?

File: src/test/java/htsjdk/samtools/sra/SRAIndexTest.java [hunk 2]
```diff
@@ -46,7 +46,7 @@
  * Created by andrii.nikitiuk on 10/28/15.
  */
 public class SRAIndexTest extends AbstractSRATest {
-    private static final SRAAccession DEFAULT_ACCESSION = new SRAAccession("SRR1298981");
+    private static final SRAAccession DEFAULT_ACCESSION = new SRAAccession("SRR2096940");
```
Comment (yfarjoun) [inline src/test/java/htsjdk/samtools/sra/SRAIndexTest.java]: why were these changed? what's wrong with the old ones?

Comment (yfarjoun) [inline src/test/java/htsjdk/samtools/sra/SRAIndexTest.java]: sorry, I see the comment now: I supposed these are the smaller files.

**Comment by yfarjoun** [comment]
@a-nikitiuk back to you. only one comment really: can you not distinguish between a bad connection and a bad accesssion string? It seems that some part of the code must know, so perhaps you can give an return code to isValid() that will be more than a boolean and work with that. The problem is that if the accession becomes "bad" because of code change we will never find out because we will skip the test.

File: src/test/java/htsjdk/samtools/sra/SRAIndexTest.java (hunk 2 above)
Comment (a-nikitiuk) [inline src/test/java/htsjdk/samtools/sra/SRAIndexTest.java]: yes, that is the reason

**Comment by a-nikitiuk** [comment]
Unfortunately, it is not possible in current version of NGS API. We are working on improvement, but that is not an easy and quick fix.

**Comment by yfarjoun** [comment]
In that case @a-nikitiuk I think it would be better to test connectivity with a single `@BeforeGroup` using an accession that is bound to exist (perhaps you have some sort of testing object that can be used for that?) and then remove the `@BeforeTest` so that each of the individual tests will fail if their accession is not valid. while this is fragile wrt losing connectivity in the middle of running the tests, it doesn't require a separate check that all the accessions are valid.

**Comment by yfarjoun** [comment]
@a-nikitiuk This needs to be rebased, and also did you have time to think about my suggestion of using `@BeforeGroup` rather than `@BeforeTest` ?

**Comment by yfarjoun** [comment]
@a-nikitiuk will you have time to rebase and replace `@BeforeMethod` with `@BeforeTest` ?

We would love to have these changes....

**Comment by a-nikitiuk** [comment]
Yeah, will do that. Sorry, did not have much time lately

**Comment by a-nikitiuk** [comment]
Rebased and added `@BeforeGroups` method

File: src/test/java/htsjdk/samtools/sra/AbstractSRATest.java [hunk 3]
```diff
@@ -4,21 +4,51 @@
...
+            Object firstParam = params[0];
+            if (firstParam instanceof String) {
+                accession = (String)firstParam;
+            } else if (firstParam instanceof SRAAccession) {
+                accession = firstParam.toString();
+            }
+        }
+
+        if (accession != null &&
+                accession.matches(SRAAccession.REMOTE_ACCESSION_PATTERN) && !canResolveNetworkAccession) {
```
Comment (yfarjoun) [inline src/test/java/htsjdk/samtools/sra/AbstractSRATest.java]: Since you are no longer asking to connect for every accession, wouldn't it be better to remove the first condition, so that if someone puts in a non-matching accession to a new test, it will fail?

Comment (a-nikitiuk) [inline src/test/java/htsjdk/samtools/sra/AbstractSRATest.java]: If someone puts a non-matching accession, then this code will do nothing. However, it detects **remote** accessions from its first argument, and if remote accession is detected and `canResolveNetworkAccession` is `false` then it skips the test.

Comment (yfarjoun) [inline src/test/java/htsjdk/samtools/sra/AbstractSRATest.java]: Oh, OK. can you then simply change the text in the Exception to:
1. note the accession that cannot be resolved, and 
2. change "SRA accession" with "remote SRA accession"?

**Comment by yfarjoun** [comment]
@a-nikitiuk are you planing to make these changes, or should I do them?

**Comment by a-nikitiuk** [comment]
You can do them if you would like to. Or I will try to find time for that today-tomorrow

**Comment by yfarjoun** [comment]
OK. I'll leave it to you then! thanks!

**Comment by a-nikitiuk** [comment]
Done

**Comment by yfarjoun** [comment]
thanks! 👍
